### ISSUE TITLE
Basis splines in formula fails in log-likelihood scoring due to array size inconsistency during normalization

### ISSUE THREAD

**Issue by Klim314** [CamDavidsonPilon/lifelines#1322]
Replication on lifelines 0.26.0

```
from lifelines.datasets import load_rossi

rossi = load_rossi()
_cph = CoxPHFitter()
_cph.fit(rossi, "week", "arrest", formula=f"bs(age, df=3, lower_bound={rossi.age.min()}, upper_bound={rossi.age.max()}) + race + wexp + mar + paro +prio")
_cph.score(rossi)

>>>ValueError: Unable to coerce to Series, length must be 7: given 8
```

This seems to be related to the initial generation of the normalized features in the score
https://github.com/CamDavidsonPilon/lifelines/blob/master/lifelines/fitters/coxph_fitter.py 2688
![image](https://user-images.githubusercontent.com/12491510/133415129-1b73f545-a997-4d9e-a54c-69b238a6d2e7.png)

comparing to the stored value of _norm_mean in the fit fitter:
```
_cph._norm_mean
>>>covariate
bs(age, df=3, lower_bound=17, upper_bound=44)[1]    0.306741
bs(age, df=3, lower_bound=17, upper_bound=44)[2]    0.146429
bs(age, df=3, lower_bound=17, upper_bound=44)[3]    0.081512
mar                                                 0.122685
paro                                                0.618056
prio                                                2.983796
race                                                0.877315
wexp                                                0.571759
```

The expansion of the mean from the bsplines was not accounted for in the dataframe normalization, leading to the error.

**Comment by CamDavidsonPilon** [comment]
Hi @Klim314 

Ah I think because `score` isn't performing the formula transform first, but it should be. Let me investigate now

**Comment by CamDavidsonPilon** [comment]
Fixed in v0.26.1
